concrete/blocks/core_area_layout/controller.php

```diff
@@ -283,41 +285,72 @@ public function save($post)
      */
     public function getImportData($blockNode, $page)
     {
+        if (!isset($blockNode->arealayout)) {
+            return [];
+        }
+        $node = $blockNode->arealayout;
+        $type = (string) $node['type'];
         $args = [];
-        if (isset($blockNode->arealayout)) {
-            $type = (string) $blockNode->arealayout['type'];
-            $node = $blockNode->arealayout;
-            switch ($type) {
-                case 'theme-grid':
-                    $args['gridType'] = 'TG';
-                    $args['arLayoutMaxColumns'] = (string) $node['columns'];
-                    $args['themeGridColumns'] = (int) (count($node->columns->column));
-                    $args['offset'] = [];
-                    $args['span'] = [];
-                    $i = 0;
-                    foreach ($node->columns->column as $column) {
-                        $args['span'][$i] = (int) ($column['span']);
-                        $args['offset'][$i] = (int) ($column['offset']);
-                        $i++;
-                    }
-                    break;
-                case 'custom':
-                    $args['gridType'] = 'FF';
-                    $args['isautomated'] = true;
-                    $args['spacing'] = (int) ($node['spacing']);
-                    $args['columns'] = (int) (count($node->columns->column));
-                    $customWidths = (int) ($node['custom-widths']);
-                    if ($customWidths == 1) {
-                        $args['isautomated'] = false;
-                    }
-                    $args['width'] = [];
-                    $i = 0;
-                    foreach ($node->columns->column as $column) {
-                        $args['width'][$i] = (int) ($column['width']);
-                        $i++;
+        switch ($type) {
+            case 'theme-grid':
+                $args += [
+                    'gridType' => 'TG',
+                    'arLayoutMaxColumns' => (string) $node['columns'],
+                    'themeGridColumns' => count($node->columns->column),
+                    'offset' => [],
+                    'span' => [],
+                ];
+                foreach ($node->columns->column as $column) {
+                    $args['span'][] = (int) $column['span'];
+                    $args['offset'][] = (int) $column['offset'];
+                }
+                break;
+            case 'custom':
```

Comment:
This doesn't seem right to me; not the logic that you have in here, I imagine that works just fine. But I don't think all of this logic should be happening under the "custom" type. These aren't custom layout types. They're preset types. I think the problem here is that export() doesn't add a new custom type for "presets" so they all go into XML as "custom" types but they shouldn't be. We should modify export() to add a new type for "preset" and then check that type in this switch() statement, instead of just grouping it all under the "custom" part of the switch.

Reply:
When exporting a preset with the migration tool, the XML is currently something like.
```xml
<arealayout type="custom" preset="theme_themehandle_presethandle>
    <!-- ... -->
</arealayout>
```

Should we update the exporter stuff so that we have something like the following?

```xml
<arealayout type="preset" preset-id="theme_themehandle_presethandle>
    <!-- ... -->
</arealayout>
```

Reply:
Yeah, something like that looks much closer than what I would expect. That would be much better for these preset layout types.

Reply:
Done in https://github.com/concrete5/concrete5/pull/10176/commits/05834082152c4fc6b900bd0b4b2d0b780c976865

Now the output for a preset is something like

```
<arealayout type="preset" preset-id="theme_themehandle_presethandle">
    <!-- ... -->
</arealayout>
```

(Just tested with a theme I'm developing)